Userland/Applications/ImageViewer/ViewWidget.cpp

```diff
@@ -77,33 +94,30 @@ void ViewWidget::navigate(Directions direction)
         }
     }
 
-    auto current_index = m_files_in_same_dir.find_first_index(m_path);
-    if (!current_index.has_value()) {
+    m_current_index = m_files_in_same_dir.find_first_index(m_path);
+    return m_current_index;
+}
+
+void ViewWidget::navigate(Directions direction)
+{
+    auto present_index = current_index();
+    if (!present_index.has_value()) {
         return;
     }
 
-    size_t index = current_index.value();
+    auto index = present_index.value();
     if (direction == Directions::Back) {
-        if (index == 0) {
-            GUI::MessageBox::show(window(), "This is the first file.", "Cannot open image", GUI::MessageBox::Type::Error);
-            return;
-        }
-
         index--;
     } else if (direction == Directions::Forward) {
-        if (index == m_files_in_same_dir.size() - 1) {
-            GUI::MessageBox::show(window(), "This is the last file.", "Cannot open image", GUI::MessageBox::Type::Error);
-            return;
-        }
-
         index++;
     } else if (direction == Directions::First) {
         index = 0;
     } else if (direction == Directions::Last) {
         index = m_files_in_same_dir.size() - 1;
     }
 
-    this->load_from_file(m_files_in_same_dir.at(index));
+    m_current_index = index;
+    this->load_from_file(m_files_in_same_dir.at(m_current_index.value()));
```

Comment:
Why not just use `index` here?
```suggestion
    this->load_from_file(m_files_in_same_dir.at(index));
```